samcli/commands/build/build_context.py

```diff
@@ -78,6 +78,7 @@ def __init__(
         locate_layer_nested: bool = False,
         hook_name: Optional[str] = None,
         build_in_source: Optional[bool] = None,
+        mount_with=MountMode.READ.value,
```

Comment:
```suggestion
        mount_with: MountMode = MountMode.READ
```

Reply:
Click.Choice takes [str] only. Workaround: Convert str to enum in BuildContext.__init__() https://github.com/aws/aws-sam-cli/blob/280b16030ddee5e1b8ff1121f39e29892a53186e/samcli/commands/build/build_context.py#L178